Unix: Add color output for console

0lvin opened the pull request:
Usecase: show map name on level load

0lvin commented:
It should also work with MacOS console, i have not tested with MacOS.

devnexen commented:
works fine on macOs too, do you think as future improvement they can be CVar ?

Yamagi commented on src/backends/unix/system.c:
```diff
@@ -107,6 +110,17 @@ Sys_Quit(void)
+		color_active = true;
+
+		Sys_ConsoleOutput("\2Terminal supports colors.\n");
+	}
```
The idea is good. However I would do it a little bit different, because checking only `xterm` only will break anything which is not xterm. For example multiplexers like screen or tmux or alternative terminals like alacritty or kitty. What about: Check if $TERM includes the substring `color` and if it does not check if $COLORTERM is set?

0lvin replied:
Thank you!

I have updated code with COLORTERM and examples what it can be in case of xfce4-terminal.

Yamagi commented:
The terminal detection logic looks good. Thank you for fixing this! However, something is still wrong with the `colorterminal` cvar. I'm starting with '+set colorterminal 0` and the game still messes with the terminal. And why is it printing everything in bold? Is that the wanted behavior or bug?

![grafik](https://user-images.githubusercontent.com/1224576/197393645-a7f141a6-e1f2-4585-898a-c08221242080.png)

0lvin commented:
> I'm starting with '+set colorterminal 0` and the game still messes with the terminal.
Do you mean that with colorterminal = 0 console have changed font style? I suppose its because cvar is initialized after init console. I will look.

> And why is it printing everything in bold? Is that the wanted behavior or bug?
Its expected behavior, code has not checked current font style, check current console style requires full termcap implementation,  and as result new white color is different to old one. I will check what I have, maybe console has not so bright colors.

0lvin commented:
I have replaced set to white to just reset font to default and removed set background to black.

As result terminals with white background should work better.

Yamagi commented:
Now it looks good. Thank you for fixing it! :)
